mpu6050_user/mpu6050.c

```diff
@@ -63,8 +63,8 @@ int gyro_temp(int i2c_fd, float *temp)
 	//printf("Val: %d\n", (short)(buf[1] | (buf[0] << 8)));
 
 	*temp = (short)(buf[1] | (buf[0] << 8));
```

Comment:
Is this conversation correct for architectures with short size 4?

Reply:
Why not, we only need 2 bytes of data temp.

Reply:
Conversation to short will do the same like conversation to int on x86_32 architecture. Can You try it?

Reply:
It will not work with 'int'.

For example:
TEMP_OUT_L = 0xC0
TEMP_OUT_H = 0xF2

For example with 'int':
(int)(TEMP_OUT_L | (TEMP_OUT_H << 8)) = 62144
Calculation: temp = data / 340 + 35
temp = 62144 / 340 + 35 = 217,776470588   ->    error, can't be 217 degrees !!!

For example with 'short':
(short)(TEMP_OUT_L | (TEMP_OUT_H << 8)) = -3392
Calculation: temp = data / 340 + 35
temp = -3392 / 340 + 35 = 25,023529412    ->     good, 25 degrees !!!

Reply:
Or I can do the conversion to int:
(int)((TEMP_OUT_L | (TEMP_OUT_H << 8)) | 0xFFFF0000)   ->    adding a minus sign

(int)((TEMP_OUT_L | (TEMP_OUT_H << 8)) | 0xFFFF0000) = -3392
Calculation: temp = data / 340 + 35
temp = -3392 / 340 + 35 = 25,023529412 -> good, 25 degrees !!!